### PR TITLE
Preventing CCData object from finding non-image extension data. 

### PR DESCRIPTION
**Pull request by anaposses**
As pointed in #7595 , if the HDU is not specified, an iteration is performed on each HDU until one is found with data. However, it is not taken into account if the HDU has image extension.

Therefore, I added in the interaction this condition. Since hdus.fileinfo() does not have any parameters that indicate the HDU extension, I looked for this information in hdus.info().

cc @Nicolopes, @RebecaMelo, @ricardogando

### CODE REVIEW

**Comment by MSeifert04** [comment]
Thanks for the pull request @anaposses.

However I don't think this works correctly. Could you add a test that ensures that this works correctly? I guess a fits file that contains a data-less primary, a table as second and the image as third HDU would be a good test candidate.

If you need help writing the test let us know.

**Comment by MSeifert04** [comment]
Not sure which milestone would be appropriate here. 2.x? or 3.0.x?

**Comment by mwcraig** [comment]
@anaposses -- let me know if you need help coming up with a file for the test @MSeifert04 suggested. Should have mentioned the need for a test in the issue...

**Comment by bsipocz** [comment]
@MSeifert04 - The code snippet in #7595 has the same behaviour with 2.0.7, so the fix should go back to LTS.

**Comment by bsipocz** [comment]
@anaposses - There are a few trailing whitespaces in a file that causes the testing suite to stop with a fail. Could you please fix them:
```
astropy/nddata/tests/test_ccddata.py:888:78: W291 trailing whitespace
astropy/nddata/tests/test_ccddata.py:889:39: W291 trailing whitespace
astropy/nddata/tests/test_ccddata.py:898:1: W391 blank line at end of file
```

Also, the PR needs a rebase as there are unrelated changes coming from the merge commit above. I'm happy to do it myself, or help you to do it.

**Comment by anaposses** [comment]
Hello @bsipocz , thank you for the response. 

I solved the trailing whitespace problem but I am still having problems with the CircleCl test.

I am trying to do so far is to add a test to test_ccddata.py in order to show that my change solves the issue. Since this is my very first PL, I am having some trouble in understanding the syntax, but I appreciate all the help you and the others gave me so far.

I suspect that the problem is in the reading of the fits image that I generated. I tried to add the file 'table_image.fits' in the 'tests/data/', and it contains:

HDU 0 - PrimaryHDU
HDU 1 - TableHDU
HDU2 -  ImageHDU

I used 'get_pkg_data_filename' to read the file, I am trying to figure out if it is right, maybe it is not.

File: astropy/nddata/tests/test_ccddata.py [hunk 1]
```diff
@@ -881,3 +882,11 @@ def test_stddevuncertainty_compat_descriptor_no_weakref():
     uncert._parent_nddata = ccd
     assert uncert.parent_nddata is ccd
     uncert._parent_nddata = None
+
+
+def test_CCDataread_returns_image():
+    # Test if CCData.read returns a image when reading a fits file containing
+    #a table and image, in that order.
+    fits_table_image = get_pkg_data_filename('data/table_image.fits', package = 'astropy.nddata.tests')
```
Comment (pllim) [inline astropy/nddata/tests/test_ccddata.py]: The `package=` part should be unnecessary. Hard to review why test failed until you have `rebase` and cleaned this up. Thanks!

Comment (anaposses) [reply astropy/nddata/tests/test_ccddata.py]: Thank you @pllim , I will do it right away!

Comment (pllim) [reply astropy/nddata/tests/test_ccddata.py]: p.s. You also seem to be committing from 3 different email addresses, two of which are not associated with your GitHub account. If you want proper GitHub author credit for all the commits, I recommend that you add the other 2 emails to your GitHub account. However, this is entirely up to you and does not affect the review of this PR. FYI.

Comment (anaposses) [reply astropy/nddata/tests/test_ccddata.py]: You are right, I had to use another computer with a different git configuration. I will fix it!

**Comment by pllim** [comment]
Something is not right with the commits/diffs... Please let us know if you need assistance in rebasing. Thanks!

**Comment by anaposses** [comment]
It passed through the test but I didn't finish the rebase yet.

**Comment by anaposses** [comment]
Hello, Since I was having too many conflicts during the rebase process I would rather close this PL and reopen with the latest commits already included.

**Pull request closed without merging**